src/assets/scripts/client/aircraft/AircraftInstanceModel.js

```diff
@@ -2356,81 +1443,65 @@ export default class Aircraft {
-            let vector;
-
-            if (this.isOnGround()) {
-                vector = vscale([sin(angle), cos(angle)], scaleSpeed);
-            } else {
-                let crab_angle = 0;
-
-                // Compensate for crosswind while tracking a fix or on ILS
-                if (this.fms.currentWaypoint.navmode === WAYPOINT_NAV_MODE.FIX || this.mode === FLIGHT_MODES.LANDING) {
-                    // TODO: this should be abstracted to a helper function
-                    const offset = angle_offset(this.heading, wind.angle + Math.PI);
-                    crab_angle = Math.asin((wind.speed * sin(offset)) / this.speed);
-                }
+        if (this.isOnGround()) {
+            vector = vscale([sin(angle), cos(angle)], trueAirSpeed);
+        } else {
+            let crab_angle = 0;
 
+            // Compensate for crosswind while tracking a fix or on ILS
+            if (this.fms.currentWaypoint.navmode === WAYPOINT_NAV_MODE.FIX || this.mode === FLIGHT_MODES.LANDING) {
                 // TODO: this should be abstracted to a helper function
-                vector = vadd(vscale(
-                    vturn(wind.angle + Math.PI),
-                    wind.speed * 0.000514444 * window.gameController.game_delta()),
-                    vscale(vturn(angle + crab_angle), scaleSpeed)
-                );
+                const offset = angle_offset(this.heading, wind.angle + Math.PI);
+                crab_angle = Math.asin((wind.speed * sin(offset)) / this.speed);
             }
 
-            this.ds = vlen(vector);
             // TODO: this should be abstracted to a helper function
-            this.groundSpeed = this.ds / 0.000514444 / window.gameController.game_delta();
-            this.groundTrack = vradial(vector);
-            this.position = vadd(this.position, vector);
-        } else {
-            this.ds = scaleSpeed;
-            this.groundSpeed = this.speed;
-            this.groundTrack = this.heading;
-            this.position = vadd(this.position, vscale([sin(angle), cos(angle)], scaleSpeed));
+            vector = vadd(vscale(
+                vturn(wind.angle + Math.PI),
+                wind.speed * 0.000514444 * window.gameController.game_delta()),
+                vscale(vturn(angle + crab_angle), trueAirSpeed)
+            );
         }
 
-        this.distance = vlen(this.position);
-        this.radial = vradial(this.position);
-
-        if (this.radial < 0) {
-            this.radial += tau();
-        }
+        this.ds = vlen(vector);
+        // TODO: this should be abstracted to a helper function
+        this.groundSpeed = this.ds / 0.000514444 / window.gameController.game_delta();
+        this.groundTrack = vradial(vector);
+        this.position = vadd(this.position, vector);
+    }
 
-        const isInsideAirspace = this.isInsideAirspace(window.airportController.airport_get());
+    /**
+     * This calculates the simplify ground speed
```

Comment:
I changed it to 
"This uses the current speed information to update the ground speed and position"

That is more clear, Any recommendations for improvement?